src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -861,26 +861,26 @@ private static bool DoesInputTypeContain(BoundExpression argument, TypeSymbol fo
             // SPEC: type or expression tree type then all the parameter types of T are
             // SPEC: input types of E with type T.
 
-            var delegateType = formalParameterType.GetDelegateType();
-            if ((object)delegateType == null)
+            var delegateOrFunctionPointerType = formalParameterType.GetDelegateOrFunctionPointerType();
+            if ((object)delegateOrFunctionPointerType == null)
             {
                 return false; // No input types.
             }
 
-            if (argument.Kind != BoundKind.UnboundLambda && argument.Kind != BoundKind.MethodGroup)
+            if (argument.Kind is not (BoundKind.UnboundLambda or BoundKind.MethodGroup or BoundKind.UnconvertedAddressOfOperator))
```

Comment:
>if (argument.Kind is not (BoundKind.UnboundLambda or BoundKind.MethodGroup or BoundKind.UnconvertedAddressOfOperator)) [](start = 12, length = 118)

It doesn't feel right to continue with ```UnconvertedAddressOfOperator``` for a delegate and with ```UnboundLambda``` for a function pointer.